### PR TITLE
Initial SMP port of sphinx drivers

### PR DESCRIPTION
**Pull request by Joshua-Anderson**

### CODE REVIEW

File: fprime-sphinx-drivers/NORFlashMgrWorker/NORFlashMgrWorkerComponentImpl.cpp [hunk 1]
```diff
@@ -376,69 +376,36 @@ namespace Drv {
       if(!Drv::isInNorAddrSpace(bank_addr)) { return NORMGR_ADDR_OUT_OF_RANGE; }
 
       // send the chip erase sequence
-  #ifdef TGT_OS_TYPE_VXWORKS
-      U32 lockKey;
-      STATUS status = taskLock();
-      FW_ASSERT(status == OK, status);
-      lockKey = intLock();
-  #endif
```
[the remark this answers is not in the document]
Comment (Joshua-Anderson) [reply fprime-sphinx-drivers/NORFlashMgrWorker/NORFlashMgrWorkerComponentImpl.cpp:384]: I've successfully run the TestNor command to reprogram the nor with the task and interrupt locks removed.

**Comment by Joshua-Anderson** [comment]
Note: The SphinxOnboardUART driver seems to sometimes panic on read, we shouldn't merge this until we resolve that issue.

**Comment by Joshua-Anderson** [comment]
A spinlock deadlock was found in the SphinxOnboardUART driver and has been corrected. SphinxOnboardUART driver is working now.

File: fprime-sphinx-drivers/Util/SphinxDrvUtil.hpp [hunk 2]
```diff
@@ -18,12 +18,20 @@
...
 #define U32_MAX (0xffffffff)
 #define I32_MAX (0x7fffffff)
 #define I32_MIN (0x80000000)
 #define BITS_PER_BYTE 8
 
 namespace Drv {
+#ifndef TGT_OS_TYPE_VXWORKS
+  /* Emulated register memory for unit tests */
+  extern std::map<U32, U32> emuRegMap;
+#endif
```
[the remark this answers is not in the document]
Comment (kevin-f-ortega) [reply fprime-sphinx-drivers/Util/SphinxDrvUtil.hpp:34]: Could you please update the README.md file? Remove the bullets under `Linux target:` and just mention `None`

**Review by kevin-f-ortega — COMMENTED** [review]
I reviewed your new updates. Looks good! Just one minor update then this should be good to merge.

**Comment by kevin-f-ortega** [comment]
@Joshua-Anderson just pinging.

**Comment by Joshua-Anderson** [comment]
@kevin-f-ortega README updated!

**Pull request merged**